server/lib/s3backup/__init__.py

```diff
@@ -0,0 +1,171 @@
+"""
+Simple module level convenience functions.
+"""
+
+import boto3
+import os
+import sys
+import glob
+import base64
+
+class Entry(object):
+    """
+    A simple class to create an entry as an object.
+    """
+
+    def __init__(self, name=None, md5=None):
+        self.name = name
+        self.md5 = md5
+
+
+    def __eq__(self, other):
+        return self.name == other.name and self.md5 == other.md5
+
+
+class s3Config(object):
+    """
+    A simple class to wrap all the configuration required to connect to s3 
+    and for the operational functions.
+    """
+
+    def __init__(self, conf):
+        # read conf from configuration file
+        self.endpoint_url = conf.get('pbench-server-backup', 'endpoint_url')
+        self.access_key_id = conf.get('pbench-server-backup', 'access_key_id')
+        self.secret_access_key = conf.get('pbench-server-backup', 'secret_access_key')
+        self.bucket_name = conf.get('pbench-server-backup', 'bucket_name')
+        self.s3 = boto3.resource('s3',
+                    aws_access_key_id='%s' % self.access_key_id,
+                    aws_secret_access_key='%s' % self.secret_access_key,
+                    endpoint_url='%s' % self.endpoint_url)
+        self.s3_client = boto3.client('s3',
```

Comment:
Once we figure out how it works exactly, we should be able to use just the resource and not worry about the client.